Applications/Spire/Include/Spire/BookView/BookViewHighlightPropertiesWidget.hpp

```diff
@@ -11,43 +14,61 @@ namespace Spire {
   class BookViewHighlightPropertiesWidget : public QWidget {
     public:
 
+      //! Signals that the recent colors have changed.
+      /*!
+        \param recent_colors The updated recent colors.
+      */
+      using RecentColorsSignal =
+        Signal<void (const RecentColors& recent_colors)>;
+
       //! Constructs a BookViewHighlightPropertiesWidget.
       /*!
         \param properties The properties to display.
+        \param recent_colors The recent colors to display.
         \param parent The parent widget.
       */
-      explicit BookViewHighlightPropertiesWidget(
-        const BookViewProperties& properties, QWidget* parent = nullptr);
+      BookViewHighlightPropertiesWidget(
+        const BookViewProperties& properties,
+        const RecentColors& recent_colors, QWidget* parent = nullptr);
 
       //! Applies the properties represented by this widget to an instance.
       /*!
         \param properties The instance to apply the properties to.
       */
       void apply(BookViewProperties& properties) const;
 
+      //! Sets the current recent colors.
+      /*!
+        \param recent_colors The current recent colors.
+      */
+      void set_recent_colors(const RecentColors& recent_colors);
```

Comment:
I don't see the purpose of this method. The get/set recent colors is a property of the ColorSelectorButton, not its parent. Almost all of these ```set_recent_colors``` methods need to be removed.

As a general principle, never change an API without getting approval first, I am open to hearing a good argument for why these should stay here but it's almost certainly going to need to be removed.

Reply:
Darryl said the recent colors are shared globally, so windows will need some way to have their recent colors updated if they're modified from another window.

In this case because the Book View properties tabs are distinct widgets, I'll still need some way to update the recent colors in the other tab's widget.

Reply:
They will be shared globally but implementing that is going to be much further down the road.

Reply:
So what functionality should this retain? Are we ignoring the recent colors completely for now?

For simplicity, I can just give the color selector drop down the default colors and remove the signals and setters for them.

Reply:
RecentColors has a public static method: ```RecentColors& get_instance();```

It should also have signals indicating when it's been updated.

Reply:
So in effect, every widget that uses RecentColors uses them independently?